src/redux/thunks/loadKubeConfig.ts

```diff
@@ -24,20 +38,34 @@ export const loadContexts = async (
         const kc = new k8s.KubeConfig();
         kc.loadFromFile(configPath);
 
-        const selectedContext = kc.contexts.find(c => c.name === currentContext);
+        let selectedContext = getSelectedContext(kc.contexts);
         if (selectedContext) {
-          kc.setCurrentContext(selectedContext && selectedContext.name);
-        } else {
-          kc.setCurrentContext((kc.contexts && kc.contexts.length > 0 && kc.contexts[0].name) || '');
+          kc.setCurrentContext(selectedContext.name);
         }
 
+        electronStore.get('kubeConfig.currentContext');
         const kubeConfig: KubeConfig = {
           contexts: kc.contexts as KubeConfigContext[],
           currentContext: kc.currentContext,
           isPathValid: kc.contexts.length > 0,
         };
 
+        kc.contexts.forEach((context) => {
+          if (!context.namespace) {
+            return;
+          }
+
+          addNamespace({
+            namespaceName: context.namespace,
+            clusterName: context.cluster,
+          });
+        });
+
         dispatch(updateProjectKubeConfig(kubeConfig));
+        dispatch(updateProjectKubeAccess(
+          getNamespaces(selectedContext?.name as string)
```

Comment:
what happens here if `selectedContext?.name` is `undefined`?

Reply:
if it's undefined kube access will be updated with an empty array